hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -315,22 +316,25 @@ AbfsThrottlingIntercept getIntercept() {
    * @return default request headers
    */
   @VisibleForTesting
-  protected List<AbfsHttpHeader> createDefaultHeaders() {
-    return createDefaultHeaders(this.xMsVersion);
-  }
+  public abstract List<AbfsHttpHeader> createDefaultHeaders();
 
   /**
    * Create request headers for Rest Operation using the specified API version.
-   * @param xMsVersion
+   * @param xMsVersion azure services API version to be used.
    * @return default request headers
    */
-  private List<AbfsHttpHeader> createDefaultHeaders(ApiVersion xMsVersion) {
-    final List<AbfsHttpHeader> requestHeaders = new ArrayList<AbfsHttpHeader>();
+  @VisibleForTesting
+  public abstract List<AbfsHttpHeader> createDefaultHeaders(ApiVersion xMsVersion);
+
+  /**
+   * Create request headers common to both service endpoints.
+   * @param xMsVersion azure services API version to be used.
+   * @return common request headers
+   */
+  protected List<AbfsHttpHeader> createCommonHeaders(ApiVersion xMsVersion) {
+    final List<AbfsHttpHeader> requestHeaders = new ArrayList<>();
     requestHeaders.add(new AbfsHttpHeader(X_MS_VERSION, xMsVersion.toString()));
-    requestHeaders.add(new AbfsHttpHeader(ACCEPT, APPLICATION_JSON
-        + COMMA + SINGLE_WHITE_SPACE + APPLICATION_OCTET_STREAM));
-    requestHeaders.add(new AbfsHttpHeader(ACCEPT_CHARSET,
-        UTF_8));
+    requestHeaders.add(new AbfsHttpHeader(ACCEPT_CHARSET, UTF_8));
```

Comment:
Why Application Json has been removed ?

Reply:
It is not removed, it is moved to DFS Client.
DFS only needs JSON, but BLOB also needs XML
Headers which need different value for diff clients will be added in respective clients.